pandas-stubs/core/series.pyi

```diff
@@ -1344,7 +1353,7 @@ class Series(IndexOpsMixin[S1], NDFrame):
         na_option: Literal["keep", "top", "bottom"] = ...,
         ascending: _bool = ...,
         pct: _bool = ...,
-    ) -> Series: ...
+    ) -> Series[float]: ...
```

Comment:
I think `rank` returns `Series[int]`

Reply:
One would assume that, but that isn't the case:

```py
>>> pd.Series([1, 2, 3, 4]).rank()
0    1.0
1    2.0
2    3.0
3    4.0
dtype: float64
```

I will add a test case for that

Reply:
Actually, it is `Series[float]` .  If there is a tie, it returns the average of the ranks.